crml/eth-bridge/src/tests.rs

```diff
@@ -342,6 +360,198 @@ fn eth_client_http_request() {
 	})
 }
 
+#[test]
+fn generate_event_proof() {
+	ExtBuilder::default().build().execute_with(|| {
+		// Test generating event proof without delay
+		let message = MockWithdrawMessage { 0: Default::default() };
+		let event_proof_id = Module::<TestRuntime>::next_proof_id();
+
+		// Generate event proof
+		assert_ok!(Module::<TestRuntime>::generate_event_proof(&message));
+		// Ensure event has not been added to delayed claims
+		assert_eq!(Module::<TestRuntime>::delayed_event_proofs(event_proof_id), None);
+		assert_eq!(Module::<TestRuntime>::next_proof_id(), event_proof_id + 1);
+		// On initialize should return 0 weight as there are no pending proofs
+		assert_eq!(
+			Module::<TestRuntime>::on_initialize(frame_system::Pallet::<TestRuntime>::block_number() + 1),
+			DbWeight::get().reads(1 as Weight)
+		);
+	});
+}
+
+#[test]
+fn delayed_event_proof() {
+	ExtBuilder::default().build().execute_with(|| {
+		let message = MockWithdrawMessage { 0: Default::default() };
+		BridgePaused::put(true);
+		assert_eq!(Module::<TestRuntime>::bridge_paused(), true);
+
+		let event_proof_id = Module::<TestRuntime>::next_proof_id();
+		let packed_event_with_id = [
+			&message.encode()[..],
+			&EthAbiCodec::encode(&Module::<TestRuntime>::validator_set().id)[..],
+			&EthAbiCodec::encode(&event_proof_id)[..],
+		]
+		.concat();
+
+		// Generate event proof
+		assert_ok!(Module::<TestRuntime>::generate_event_proof(&message));
+		// Ensure event has been added to delayed claims
+		assert_eq!(
+			Module::<TestRuntime>::delayed_event_proofs(event_proof_id),
+			Some(packed_event_with_id)
+		);
+		assert_eq!(Module::<TestRuntime>::next_proof_id(), event_proof_id + 1);
+
+		// Re-enable bridge
+		BridgePaused::put(false);
+		// initialize pallet and initiate event proof
+		let max_delayed_events = Module::<TestRuntime>::delayed_event_proofs_per_block() as u64;
+		let expected_weight: Weight =
+			DbWeight::get().reads(3 as Weight) + DbWeight::get().writes(2 as Weight) * max_delayed_events;
+		assert_eq!(
+			Module::<TestRuntime>::on_initialize(frame_system::Pallet::<TestRuntime>::block_number() + 1),
+			expected_weight
+		);
+		// Ensure event has been removed from delayed claims
+		assert_eq!(Module::<TestRuntime>::delayed_event_proofs(event_proof_id), None);
+	});
+}
+
+#[test]
+fn multiple_delayed_event_proof() {
+	ExtBuilder::default().build().execute_with(|| {
+		let message = MockWithdrawMessage { 0: Default::default() };
+		BridgePaused::put(true);
+		assert_eq!(Module::<TestRuntime>::bridge_paused(), true);
+
+		let max_delayed_events = Module::<TestRuntime>::delayed_event_proofs_per_block();
+		let event_count: u8 = max_delayed_events * 2;
+		let mut event_ids: Vec<EventProofId> = vec![];
+
+		for _ in 0..event_count {
+			let event_proof_id = Module::<TestRuntime>::next_proof_id();
+			event_ids.push(event_proof_id);
+			let packed_event_with_id = [
+				&message.encode()[..],
```

Comment:
would be good to change message value so its different in storage
then check the messages left in/out of storage are what's expected